### PR TITLE
Configurable retry for errors

### PR DESCRIPTION
**Pull request by aggFTW**

### CODE REVIEW

File: sparkmagic/sparkmagic/livyclientlib/configurableretrypolicy.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        super(ConfigurableRetryPolicy, self).__init__(-1, max_retries)
+
+        # If user configured to an empty list, let's make this behave as
+        # a Linear Retry Policy by assigning a list of 1 element.
+        if len(retry_seconds_to_sleep_list) == 0:
+            retry_seconds_to_sleep_list = [5]
+        elif not all(n > 0 for n in retry_seconds_to_sleep_list):
+            raise BadUserConfigurationException(u"All items in the list in your config need to be positive for configurable retry policy")
+
+        self.retry_seconds_to_sleep_list = retry_seconds_to_sleep_list
```
Comment (CodingCat) [inline sparkmagic/sparkmagic/livyclientlib/configurableretrypolicy.py]: to make things simple, shall we implement exponential backoff and just let the user setup a threshold?

http://docs.aws.amazon.com/general/latest/gr/api-retries.html

File: sparkmagic/sparkmagic/livyclientlib/linearretrypolicy.py [hunk 2]
```diff
@@ -8,12 +8,12 @@ class LinearRetryPolicy(object):
 
     def __init__(self, seconds_to_sleep, max_retries):
         self._seconds_to_sleep = seconds_to_sleep
-        self._max_retries = max_retries
```
Comment (CodingCat) [inline sparkmagic/sparkmagic/livyclientlib/linearretrypolicy.py]: what's the difference here? starting with a `_` indicates this is for privately usage?

Comment (aggFTW) [inline sparkmagic/sparkmagic/livyclientlib/linearretrypolicy.py]: Yes, python convention.

File: sparkmagic/sparkmagic/livyclientlib/configurableretrypolicy.py (hunk 1 above)
Comment (aggFTW) [inline sparkmagic/sparkmagic/livyclientlib/configurableretrypolicy.py]: We can also implement this on top or like in the code you posted above.

**Comment by aggFTW** [comment]
Thanks for the review @CodingCat!

**Pull request merged**